src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +158,123 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
     }
 }
 
+void IInferRequestInternal::SetBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    OPENVINO_ASSERT(
+        !std::any_of(blobs.begin(),
+                     blobs.end(),
+                     [](const Blob::Ptr& item) {
+                         return !item || item->is<RemoteBlob>() || item->is<CompoundBlob>();
+                     }),
+        "set_input_tensors/set_tensors error. Default implementation doesn't support RemoteTensor or legacy "
+        "CompoundBlob objects");
+
+    OPENVINO_ASSERT(!blobs.empty(),
+                    "set_input_tensors/set_tensors can't be called with empty blobs for input '",
+                    name,
+                    "'");
+    if (blobs.size() == 1) {
+        SetBlob(name, blobs[0]);
+        return;
+    }
+    std::shared_ptr<const ov::op::v0::Parameter> param;
+    const auto& inputs = GetInputs();
+    for (const auto& input : inputs) {
+        if (auto p = std::dynamic_pointer_cast<const ov::op::v0::Parameter>(input)) {
+            const auto& names = p->output(0).get_tensor().get_names();
+            if (names.find(name) != names.end()) {
```

Comment:
you are trying to find name as tensor name, while IE 1.0 operates with operation names, so `name` should be an operation name